autocorpus/table.py

```diff
@@ -6,565 +6,583 @@
 from pathlib import Path
 from typing import Any
 
-from .utils import get_data_element_node, handle_tables, navigate_contents
-
+from bs4 import BeautifulSoup, NavigableString, Tag
 
-class Table:
-    """Table-JSON builder class."""
+from .utils import get_data_element_node, handle_tables, navigate_contents
 
-    def __table_to_2d(self, t, config):
-        """Transform tables from nested lists to JSON.
+pval_regex = r"((\d+\.\d+)|(\d+))(\s?)[*××xX](\s{0,1})10[_]{0,1}([–−-])(\d+)"
+pval_scientific_regex = r"((\d+.\d+)|(\d+))(\s{0,1})[eE](\s{0,1})([–−-])(\s{0,1})(\d+)"
+
+
+def __table_to_2d(t: BeautifulSoup) -> list[list[str]] | None:
+    """Transform tables from nested lists to JSON.
+
+    Args:
+        t (bs4.BeautifulSoup): html table, beautiful soup object
+
+    Returns:
+        (list): table in JSON format
+
```

Comment:
We don't need the type hints in the docstring too:

```suggestion
    Args:
        t: HTML table

    Returns:
        Table in JSON format
```

Unrelated, but I feel like the term JSON is a bit misleading here